lib/cib/cib_file.c

```diff
@@ -188,6 +196,29 @@ static gboolean cib_do_chown = FALSE;
                                                 #flags_to_clear);       \
     } while (0)
 
+static int
+get_operation(const char *op, const cib_operation_t **operation)
+{
+    if (op_table == NULL) {
+        op_table = pcmk__strkey_table(NULL, NULL);
+
+        for (int lpc = 0; lpc < PCMK__NELEM(cib_file_ops); lpc++) {
+            const cib_operation_t *oper = &(cib_file_ops[lpc]);
+
+            g_hash_table_insert(op_table, (gpointer) oper->name,
+                                (gpointer) oper);
+        }
+    }
+
+    *operation = g_hash_table_lookup(op_table, op);
+
+    if (*operation == NULL) {
+        crm_err("Operation %s is not valid", op);
+        return -EINVAL;
```

Comment:
EOPNOTSUPP or EPROTO might be better

Reply:
My main thought is EINVAL maps to CRM_EX_SOFTWARE (i.e. bug) as exit code, while EOPNOTSUPP is CRM_EX_UNIMPLEMENT_FEATURE and EPROTO is CRM_EX_PROTOCOL. It might never be mapped to an exit status, but it still gives some logical direction. In this case, really any of the three would make logical sense, so I'm fine with any.